### PR TITLE
Add support for node packages

### PR DESCRIPTION
**Pull request by scothis**
A node package is distinguished from a node module by a package.json
file. A package can contain one or more modules. The package.json file
provides a wealth of metadata including the dependencies to be
installed, the main entry module into the package, custom scripts, and
more.

When a package.json file is present these key changes should occur:
1. all files in the directory are copied into the container
2. production dependencies are installed via npm in the docker daemon
3. the entry point should be defined by the package, and the invoker
   should require the directory instead of a specific module

The package.json file can also provide a conveient hook to apply source
transformations via a postinstall script. For example, compiling
TypeScript, or ES.next via Babel.

Closes projectriff/node-function-invoker#19

### CODE REVIEW

**Comment by trisberg** [comment]
The "delete" command changes shouldn't change the use of a single file with `-f` option

```
riff delete --namespace default --dry-run -f ./square-function.yaml 

Delete Command: kubectl delete --namespace default
```
That should have been:
```
Delete Command: kubectl delete --namespace default -f /Users/trisberg/Test/riff/square/square-function.yaml
```

The apply command has similar issue although the current version isn't correct either, see #37

File: pkg/initializers/node/dockerfile_generator.go [hunk 1]
```diff
@@ -41,20 +41,29 @@ ADD {{.ArtifactBase}} ${FUNCTION_URI}
 {{- end }}
 `
 
-// TODO create .dockerignore file
 var nodeFunctionDockerIgnoreTemplate = `
-{{ if .PackageJSONExists -}}
-node_modules
-{{- end }}
-`
+ {{ if .PackageJSONExists -}}
```
Comment (trisberg) [inline pkg/initializers/node/dockerfile_generator.go]: shouldn't have a leading space

**Comment by scothis** [comment]
@trisberg thanks for the feedback. Rebased on top of master and added a commit addressing your comments

**Comment by trisberg** [comment]
Squashed, rebased and merged as commit b823f01878cd0ffff9a320c4eb0260c5e1cc13bc

**Pull request closed without merging**